code/extensions/CommentsExtension.php

```diff
@@ -480,6 +480,22 @@ public function CommentsForm()
             ->renderWith('CommentsInterface');
     }
 
+    /*
+    JavaScript form reply requires an empty form, when the reply button is
+    clicked the correct values are inserted, the form moved, and then shown
+     */
+    public function getReplyFormForJavaScript() {
+        // Build reply controller
+        $controller = CommentingController::create();
+        $controller->setOwnerRecord($this->owner);
+        $controller->setBaseClass($this->owner->ClassName);
+        $controller->setOwnerController(Controller::curr());
+        $comment = new Comment();
+        $comment->ID='JSOnly';
```

Comment:
This code is blocking merge; I'm not happy with setting comments with invalid ID's as "dummy" placeholders for including javascript.

Please see the code example I linked earlier for an optional alternative.